MantidPlot/CMakeLists.txt

```diff
@@ -823,6 +823,10 @@ target_include_directories(MantidPlot PRIVATE
 target_compile_options( MantidPlot PRIVATE
   $<$<AND:$<CXX_COMPILER_ID:GNU>,$<VERSION_GREATER_EQUAL:$<CXX_COMPILER_VERSION>,8.0>>:-Wno-cast-function-type> )
 
+set (BOOST_LIBRARIES_MANTIDPLOT ${Boost_LIBRARIES})
+find_package(BoostPython REQUIRED)
```

Comment:
I've noticed a number of CMakelists.txt are doing this in Mantid, including our Framework/PythonInterface.

As it's required for the entire project at this point could you move this to `/CMakeLists.txt` with our other mandatory deps, I'll create a maintenance issue to clean up this find_package everywhere else.

After we can drop everything here....

Reply:
So I tried this and got a collection of linker errors, let's make dealing with this something for another PR

Reply:
BoostPython is still only required by a subset of libraries: The PythonInterface ones, `MantidPlot` and `mantidqt`. If we move it can we make sure we still keep the linking to only those libraries that are currently using it? I think what we'll find is that `find_package(BoostPython REQUIRED)` clears and overwrites `Boost_LIBRARIES` so that's where the linker errors come from. We'll need to find the standard boost libraries, cache the value, call `find_package(BoostPython REQUIRED)`, set a new `BoostPython_LIBRARIES` variable to its result and then restore `Boost_LIBRARIES`

Reply:
>If we move it can we make sure we still keep the linking to only those libraries that are currently using it?

My naive assumption is yes, provided we don't add it as a global library and do it on a target by target basis we will only link it to the targets that need it.

An advantage of this a single `find_package` is avoiding people getting it wrong (such as forgetting to restore `Boost_LIBRARIES`) and constantly re-searching for it.

> I think what we'll find is that find_package(BoostPython REQUIRED) clears and overwrites Boost_LIBRARIES so that's where the linker errors come from. 

Ahh, I was scratching my head on this one. I've opened #28431 to do this work in a future maintenance period. But I think it's fine as in for this PR